### PR TITLE
Enable WebAssembly target in LLVM backend by default

### PR DESCRIPTION
**Pull request by kateinoigakukun**
Enable WebAssembly target in LLVM backend for OSS toolchain packages since we merged all compiler fixes for Wasm target. This unlocks further CI setup for standalone stdlib build.

It seems there is no significant binary-size/build-time increase.

| | Before | After |
|:-:|:-:|:-:|
| Toolchain Size (unpacked) | 2.173 GB[^1] | 2.186 GB[^2] (+ 13 MB) |
| Toolchain Size (packed) | 667 MB  | 671 MB (+ 4 MB) |
| Toolchain Build-time | 1 hr 12 min [^3] | 1 hr 14 min [^4]|

[^1]: https://download.swift.org/tmp/pull-request/66539/594/ubuntu2004/PR-ubuntu2004.tar.gz
[^2]: https://download.swift.org/tmp/pull-request/68750/595/ubuntu2004/PR-ubuntu2004.tar.gz
[^3]: https://ci.swift.org/job/swift-PR-toolchain-Linux/594/
[^4]: https://ci.swift.org/job/swift-PR-toolchain-Linux/595/

### CODE REVIEW

**Comment by kateinoigakukun** [comment]
@swift-ci Please Build Toolchain Linux Platform

**Comment by kateinoigakukun** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by kateinoigakukun** [comment]
@swift-ci Please smoke test

**Comment by kateinoigakukun** [comment]
@swift-ci Please smoke test macOS platform

**Review by etcwilde — DISMISSED** [review]

**Review by etcwilde — COMMENTED** [review]
The change itself looks fine and thank you for including the size/build time information.
What tests are enabled for WASM? I'd like to make sure that things are working as expected and will continue working once this is enabled.

**Comment by MaxDesiatov** [comment]
> To ensure a produced Wasm binary "works well", we need stdlib and runtime build, and also CI environment setup to install WebAssembly runtime, and this change unblocks them.

+1 to this point. If enabling stdlib and runtime build tests on CI is a prerequisite for this change, then we'll be stuck in a circular logic. Those tests can't be enabled or even tested locally off the code available in this repository without enabling the LLVM backend for this platform in the first place.

**Comment by kateinoigakukun** [comment]
@etcwilde @MaxDesiatov The backend will be tested enough after merging https://github.com/apple/swift/pull/69101
Do you think it's ready to merge this after the PR will be merged?

**Comment by kateinoigakukun** [comment]
Please test with following pull request:
https://github.com/apple/swift-integration-tests/pull/122

@swift-ci Please smoke test

**Comment by kateinoigakukun** [comment]
Please test with following pull request:
https://github.com/apple/swift-integration-tests/pull/122

@swift-ci Please build toolchain

**Comment by kateinoigakukun** [comment]
Please test with following pull request:
https://github.com/apple/swift-integration-tests/pull/122

@swift-ci Please build toolchain Windows platform

**Comment by kateinoigakukun** [comment]
@swift-ci Please test Windows platform

**Comment by shahmishal** [comment]
```
PASS: swift-package-tests :: test-cross-compile/test-compile-wasm.test (12 of 35)
```

**Review by shahmishal — APPROVED** [review]

**Pull request merged**
